### DISCUSSION TITLE
How to animate big object(1920x1080) background

### DISCUSSION THREAD

**Discussion by jacobdab** [pixijs/pixijs#7967] [Q&A]
Hey, 

Making 1920x1080 background png sequence would make 100+ pngs spritesheets. WebM(VP9) makes smooths backgrounds for 1920x1080, but for some reason I didn't see anybody mentioning it. From what I know webM(vp9) can brings some troubles on iOS devices... Mp4 would solve them ,but it doesn't have alpha channel and of course native .avi or .mov takes too much space. 

What is the best way to animate 1920x1080 or 2048x1080 background with pixi.JS?

**Comment by eXponenta** [comment]
Only video.
Because you cannot store more that 10 frames in GPU - it will be expensive.

You can use Basis for streamed decompression, but i think for compression of 100 frames with alpha, you will wait about 10+ hours :) 
https://github.com/BinomialLLC/basis_universal

(haha, demo link is gone)

You can use 2 mp4, rgb + alpha, but should be strictly encode it with same bitrate to avoid time-shifts.
And then mix via shader.

Or use a video where a one half of frame is RGB, other half is alpha, and mix it shader too.

Or stupidly use a chromakey and forget about partial transparency :)

**Comment by jacobdab** [comment]
How I can use 2 mp4? What do you use to make rgb mp4 and alpha mp4?

**Reply by themoonrat** [reply]
Have you considered having the mp4 background videos done in html rather than PixiJS? That way you avoid the WebGL performance penalty of constantly uploading and decoding video frames?

Otherwise, there's no easy & good solution for having very large frames of animation, so you need to think about alternatives. Could parts of the background be static, but small section animate instead? Or think about using something like Spine, which can let artists have a lot of control over a scene, and animate it in an optimal way, so only the parts that need to have separate frames do.

**Reply by jacobdab** [reply]
I read somewhere that using PIXI.Texture.from(webmVideo) is using html video element as default.

**Reply by bigtimebuddy** [reply]
What @themoonrat is saying is that you layer PixiJS canvas _on-top_ of a video element  (`<video>`) on your page. So you end up controlling the video outside Pixi's rendering. While Pixi can manage your HTMLVideo element, there's a serious performance cost having to upload to the GPU, better to keep these separate and the user will never know the difference.

**Reply by eXponenta** [reply]
Topicstarter require a transparent video, looks like that it will using with some composition elements.
Maybe this is a like water: there are a front face with boat and back face with fish that should be blended between.

@jacobdab can you provide what you real require?
Maybe create a minimal demo or draw some schema.
